tests/unit/TranslateTest.js

```diff
@@ -97,3 +94,45 @@ describe('Translation Keys', () => {
         });
     });
 });
+
+describe('flattenObject', () => {
+    it('It should work correctly', () => {
+        const func = ({content}) => `This is the content: ${content}`;
+        const simpleObject = {
+            common: {
+                yes: 'Yes',
+                no: 'No',
+            },
+            complex: {
+                activity: {
+                    none: 'No Activity',
+                    some: 'Some Activity',
+                },
+                report: {
+                    title: {
+                        expense: 'Expense',
+                        task: 'Task',
+                    },
+                    description: {
+                        none: 'No description',
+                    },
+                    content: func,
+                    messages: ['Hello', 'Hi', 'Sup!'],
+                },
+            },
+        };
+
+        const result = translations.flattenObject(simpleObject);
+        expect(result).toStrictEqual({
```

Comment:
@BeeMargarida why do we check the results?
Does it support the deep equality in objects?

Reply:
Yes, we are checking because this is a unit test, and strictEqual is a check with deep equality, it checks values as well as structure